api/client/client.go

```diff
@@ -0,0 +1,249 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/base64"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/api/grpc/disperser"
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigenda/core/auth"
+	"github.com/Layr-Labs/eigenda/encoding"
+	"github.com/Layr-Labs/eigenda/encoding/utils/codec"
+	"github.com/ethereum/go-ethereum/log"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+type BlobEncodingVersion byte
+
+var NoIFFT BlobEncodingVersion = 0x00
+
+type IEigenDAClient interface {
+	GetBlob(ctx context.Context, BatchHeaderHash []byte, BlobIndex uint32) ([]byte, error)
```

Comment:
Will this only support retrieval through dispersal endpoint or also support decentralized retrieval from DA nodes?
Retrieval from DA nodes will require additional parameters

Reply:
The goal of this PR is to unify logic that exists today in several rollup integrations. Adding support for retriever sidecars is a more involved effort that will require operator nodes to stop whitelisting the disperser IP.